packages/xdl/src/modules/config.ts

```diff
@@ -578,34 +593,32 @@ const vendoredNativeModules = [
   { libName: 'react-native-webview', sdkVersions: '>=33.0.0', isNativeModule: true },
 ];
 
-function defaults(defaultConfig, ...customConfigs) {
-  const config = { ...defaultConfig };
-  for (const customConfig of customConfigs) {
-    if (customConfig) {
-      Object.assign(config, customConfig || {});
-    }
-  }
-  return config;
+function defaults(
+  defaultConfig: NativeConfig,
+  ...customConfigs: (Partial<NativeConfig> | undefined)[]
+) {
+  return Object.assign({}, defaultConfig, ...customConfigs);
 }
 
-const expoUniversalModules = [...universalModules, ...expoSdkUniversalModules];
-
-const expoSdkUniversalModulesConfigs = expoUniversalModules.map(
-  ({ config, podName, libName, sdkVersions, isNativeModule, ...params }) => {
+export const expoSdkUniversalModulesConfigs: ModuleConfig[] = expoUniversalModules.map(
+  ({ config = {}, podName, libName, sdkVersions, isNativeModule = true, ...params }) => {
     return {
       podName,
       libName,
       sdkVersions,
-      isNativeModule: isNativeModule == null ? true : isNativeModule,
+      isNativeModule,
       config: {
-        ios: defaults(defaultUniversalModuleConfig.ios, params, config && config.ios),
-        android: defaults(defaultUniversalModuleConfig.android, params, config && config.android),
+        ios: defaults(
+          defaultUniversalModuleConfig.ios,
+          params,
+          'ios' in config ? config.ios : undefined
```

Comment:
What's the difference between `'ios' in config ? config.ios : undefined` and `config.ios`?
Also, according to the type you defined (https://github.com/expo/expo-cli/pull/719/files#diff-f18029ee47d1618e49360d08f9fe307dR8) `config.ios` key should never be `undefined`, hmm?

Reply:
The `'ios' in config` guard is there to satisfy the compiler, which doesn't allow indexing undefined properties.

This function converts partial configuration objects to the `ModuleConfig` type. The `ios` property can be undefined before the object is processed by this function. Then this function sets defaults for any missing values (e.g. it uses `defaultUniversalModuleConfig.ios` as the default value for the `ios` property).